src/api/helpers/formatWorkData.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2019  Akhilesh Kumar
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+/* eslint-disable */
+import _ from 'lodash';
+
+export function getWorkBasicInfo(work: object) {
+	return _.isNil(work) ? null :
+		{
+			bbid: work.bbid,
+			defaultAlias: {
+				name: work.defaultAlias.name,
```

Comment:
Here and everywhere else where accessing a nested property (that can throw an uncaught error like  "cannot find property 'name' of undefined"), please use lodash's _.get method which does so safely (without throwing an error).

You should also use the same method to always return a default value ([`_.get(object, "path", default)`](https://lodash.com/docs/4.17.11#get)) that the client can expect (for example `_.get(work, 'disambiguation.comment', null)`)
For example, for languages, `languages: _.get(work, 'languageSet.languages', []).map((language) => language.name)` will ensure that you can use .map , and that you always return an array (empty array in this case) so that the consumer knows it is safe to use array methods when using results from the api.